client/src/plus/collections/v2/api.ts

```diff
@@ -1,76 +1,234 @@
+  const response = await fetch(key, {
+    body: JSON.stringify(body),
+    method: "POST",
+    headers: {
+      "content-type": "application/json",
+    },
+  });
+  if (!response.ok) throw Error(response.statusText);
+  try {
+    return await response.json();
+  } catch {
+    return response;
   }
-  return collections;
 }
 
-export async function getCollection(
-  id: string | undefined
-): Promise<Collection | undefined> {
-  if (!id) throw Error;
-  return (
-    JSON.parse(
-      window.localStorage.getItem("collections") || "[]"
-    ) as Collection[]
-  ).find((c) => c.id === id);
+async function deleter(key: string | undefined) {
+  if (!key) throw Error("Invalid key");
+  return fetch(key, {
+    method: "DELETE",
+  });
 }
 
-export async function getItem(doc: Doc): Promise<Item | undefined> {
-  return (
-    JSON.parse(window.localStorage.getItem("items") || "[]") as Item[]
-  ).find((b) => b.url === doc.mdn_url);
+export function useCollections() {
+  return useSWR<Collection[]>(
+    COLLECTIONS_ENDPOINT,
+    async (key: string) => await fetcher<MultipleCollectionInfo[]>(key)
+  );
 }
 
-export async function saveItem(item: Item): Promise<void> {
-  const other_items = (
-    JSON.parse(window.localStorage.getItem("items") || "[]") as Item[]
-  ).filter((b) => b.url !== item.url);
-  window.localStorage.setItem("items", JSON.stringify([...other_items, item]));
+export function useCollection(id: string | undefined) {
+  return useSWR<Collection>(
+    getCollectionKey(id),
+    async (key: string) => await fetcher<MultipleCollectionResponse>(key)
+  );
 }
 
-export async function deleteItem(item: Item): Promise<void> {
-  const items = (
-    JSON.parse(window.localStorage.getItem("items") || "[]") as Item[]
-  ).filter((b) => b.url !== item.url);
-  window.localStorage.setItem("items", JSON.stringify(items));
+export async function addCollection(
+  collection: NewCollection
+): Promise<Collection> {
+  const response = await poster<
+    MultipleCollectionCreationRequest,
+    MultipleCollectionInfo
+  >(COLLECTIONS_ENDPOINT, collection);
+  mutate(COLLECTIONS_ENDPOINT);
+  return response;
 }
 
-export async function getItems(collection: Collection): Promise<Item[]> {
-  return (
-    JSON.parse(window.localStorage.getItem("items") || "[]") as Item[]
-  ).filter((b) => b.collection_id === collection.id);
+export async function editCollection(
+  collection: Collection
+): Promise<Collection> {
+  const response = await poster<
+    MultipleCollectionCreationRequest,
+    MultipleCollectionInfo
+  >(getCollectionKey(collection.id), collection);
+  mutate(COLLECTIONS_ENDPOINT);
+  mutate(getCollectionKey(collection.id));
+  return response;
+}
+
+export async function deleteCollection(
+  collection: Collection
+): Promise<Response> {
+  const { id } = collection;
+  const response = await deleter(getCollectionKey(id));
+  mutate(COLLECTIONS_ENDPOINT);
+  mutate(getCollectionKey(id));
+  // TODO: mutate(getBookmarkKey());
```

Comment:
Like my other comment - this was here to try and keep the state completely up to date - but commented out, because we don't know which bookmarks are in this collection at this point. I'll remove.